runtime/auth/aws-signing-tests/common/src/aws/smithy/kotlin/runtime/auth/awssigning/tests/AwsChunkedTestBase.kt

```diff
@@ -365,9 +393,9 @@ abstract class AwsChunkedTestBase(
         val data = ByteArray(dataLengthBytes) { 0x7A.toByte() }
         var previousSignature: ByteArray = byteArrayOf()
 
-        val trailingHeaders = Headers {
-            append("x-amz-checksum-crc32", "AAAAAA==")
-            append("x-amz-arbitrary-header-with-value", "UMM")
+        val trailingHeaders = DeferredHeaders {
+            append("x-amz-checksum-crc32", CompletableDeferred("AAAAAA=="))
+            append("x-amz-arbitrary-header-with-value", CompletableDeferred("UMM"))
         }
```

Comment:
@aajtodd I don't understand your comment, passing a `String` value to the `CompletableDeferred` constructor marks it as completed, so I don't see how it would be useful to return it.

Reply:
Right, I see. I think my concern here was if we are exposing `CompletableDeferred` instances from the `DeferredHeaders` type (e.g. via `get`). The "complete" side of a deferred should be retained by the producer and not be accessible by a consumer of `DeferredHeaders`. As long as that is the case we're good.